walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,97 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request.getDuration() > 1440)
+      return Collections.emptyList();
+
+    // TreeSet has O(log n) insertion time
```

Comment:
This comment should be more detailed. Why is log n time significant, and what does it enable your algorithm to do? This is especially important given that TreeSet iterates over its values in a sorted order, which is non-obvious to folks reading the code.